Mappings and Explanatory Javadoc for Chunk Update Changes

shartte opened the pull request:
There were some changes to chunk update since 1.16 and this PR tries to:

- Add some of the mappings related to it
- Add explanatory Javadoc where possible to make it easier for other people poking through that code

liach commented on mappings/net/minecraft/network/packet/s2c/play/ChunkDataS2CPacket.mapping:
```diff
@@ -7,9 +7,16 @@ CLASS net/minecraft/class_2672 net/minecraft/network/packet/s2c/play/ChunkDataS2
 	FIELD field_12238 blockEntities Ljava/util/List;
 	FIELD field_16416 heightmaps Lnet/minecraft/class_2487;
 	FIELD field_20664 biomeArray [I
+	FIELD field_25720 skipLightingUpdate Z
```
`retainOldLighting`?

liach commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +75,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 	METHOD method_22446 stream (Lnet/minecraft/class_1923;I)Ljava/util/stream/Stream;
 		ARG 0 center
 		ARG 1 radius
+	METHOD method_30551 getPackedLocalPosX (S)I
```
`unpack(To)LocalX`, so on

shartte replied:
That would be inconsistent with the other methods, although I also like that more :-|
Also, "getPacked..." should be "packLocalPos", would you also favor that change together with renaming all of these to "unpack..."?

shartte replied:
Oh and also inconsistent with the functions to pack/unpack the position of chunk positions themselves.

liach replied:
Yeah `getPacked` is accessing a packed value. However there is no packed value to access here: the packed value is rather passed by the method caller and the unpacked value is returned. Hence `unpack` than `getPacked`, and other `getPacked` methods need fixing as well If they are unpacking a value than accessing a packed value.

shartte replied:
No, what I mean the other method is actually doing the packing :-D
But I take that as a yes.

shartte replied:
On second thought, I looked through the methods again and I'd rather leave them as is. The class has a shitton of methods dealing with various different coordinate systems, and I think being verbose here - while tedious - avoids mistakes.

liach replied:
So this method actually does packing? Are you kidding me, packing a short into an integer?

shartte replied:
No, but the alternative is: `getXFromPackedLocalPos`.
Regarding pack/unpack: It's not consistent with any of the other methods in this class that do packing.
I.e. `asLong`...

shartte replied:
Okay, I refactored more members in this class and added alot more docs. Is this better?

liach commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +75,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT Gets the local z-coordinate from the given packed local position.
+		COMMENT @see #getPackedLocalPos
+		ARG 0 packedLocalPos
+	METHOD method_30554 getPackedLocalPosBlockX (S)I
```
`unpack(To)WorldX`, so on

shartte replied:
See my comment above. I'd rather keep these verbose names.

liach reviewed (changes requested):
Unfortunately I refuse to approve this pr until we can sort out this pack and unpack mess. Current names make me feel like the returned values of those methods are packed, while it most likely is not packed since you don't pack a short into an integer.

liach reviewed (approved):
This looks so much better

YanisBft commented on mappings/net/minecraft/network/packet/s2c/play/ChunkDataS2CPacket.mapping:
```diff
@@ -25,3 +32,6 @@ CLASS net/minecraft/class_2672 net/minecraft/network/packet/s2c/play/ChunkDataS2
 	METHOD method_11530 isFullChunk ()Z
 	METHOD method_16123 getHeightmaps ()Lnet/minecraft/class_2487;
 	METHOD method_22422 getBiomeArray ()[I
+	METHOD method_30144 isSkipLightingUpdate ()Z
```
```suggestion
	METHOD method_30144 shouldSkipLightingUpdate ()Z
```

YanisBft commented on mappings/net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket.mapping:
```diff
@@ -1 +1,16 @@
+	FIELD field_26347 blockState [Lnet/minecraft/class_2680;
+	METHOD <init> (Lnet/minecraft/class_4076;Lit/unimi/dsi/fastutil/shorts/ShortSet;Lnet/minecraft/class_2826;)V
+		ARG 1 sectionPos
+			COMMENT The position of the given chunk section that will be sent to the client.
```
```suggestion
			COMMENT the position of the given chunk section that will be sent to the client
```

YanisBft commented on mappings/net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket.mapping:
```diff
@@ -1 +1,16 @@
+		ARG 1 sectionPos
+			COMMENT The position of the given chunk section that will be sent to the client.
+		ARG 2 updatedLocalPosSet
+			COMMENT The set of packed local positions within the given chunk section that should be included in the packet.
```
```suggestion
			COMMENT the set of packed local positions within the given chunk section that should be included in the packet
```

YanisBft commented on mappings/net/minecraft/server/world/ChunkHolder.mapping:
```diff
@@ -20,6 +20,13 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
+		COMMENT Contains the packed chunk-local positions that have been marked for update
+		COMMENT by {@link #markForBlockUpdate}, grouped by their chunk y-section.
+		COMMENT <p>
+		COMMENT Entries for a section are null, if the section has no positions marked for update.
```
```suggestion
		COMMENT Entries for a section are null if the section has no positions marked for update.
```

YanisBft commented on mappings/net/minecraft/server/world/ChunkHolder.mapping:
```diff
@@ -50,6 +57,7 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
 	METHOD method_14012 markForLightUpdate (Lnet/minecraft/class_1944;I)V
 		ARG 1 type
 		ARG 2 y
+			COMMENT Chunk section y coordinate
```
```suggestion
			COMMENT chunk section y coordinate
```

haykam821 replied:
```suggestion
			COMMENT Chunk section Y coordinate
```

YanisBft commented on mappings/net/minecraft/server/world/ChunkHolder.mapping:
```diff
@@ -63,14 +71,18 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
 		METHOD method_14014 isAfter (Lnet/minecraft/class_3193$class_3194;)Z
 			ARG 1 levelType
 	CLASS class_3724 Unloaded
+		COMMENT This is used to represent a chunk that has not been loaded yet.
```
```suggestion
		COMMENT Used to represent a chunk that has not been loaded yet.
```

liach replied:
```suggestion
		COMMENT Represents a chunk that has not been loaded yet.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 pos
 	METHOD method_18683 getSectionY ()I
 	METHOD method_18684 getLocalCoord (I)I
+		COMMENT Converts a world-coordinate to the local coordinate system (0-15) of its corresponding chunk section.
```
```suggestion
		COMMENT Converts a world coordinate to the local coordinate system (0-15) of its corresponding chunk section.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 3 z
 	METHOD method_18674 getSectionX ()I
 	METHOD method_18675 getSectionCoord (I)I
+		COMMENT Converts a world-coordinate to the corresponding chunk-section coordinate.
```
```suggestion
		COMMENT Converts a world coordinate to the corresponding chunk section coordinate.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT Converts a world-coordinate to the corresponding chunk-section coordinate.
 		ARG 0 coord
 	METHOD method_18676 from (III)Lnet/minecraft/class_4076;
+		COMMENT Creates a chunk section position from its x, y and z coordinates.
```
```suggestion
		COMMENT Creates a chunk section position from its X, Y, and Z coordinates.
```

haykam821 commented on mappings/net/minecraft/server/world/ChunkHolder.mapping:
```diff
@@ -20,6 +20,13 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
+		COMMENT Indicates that {@link #blockUpdatesBySection} contains at least one entry.
+	FIELD field_25804 blockUpdatesBySection [Lit/unimi/dsi/fastutil/shorts/ShortSet;
+		COMMENT Contains the packed chunk-local positions that have been marked for update
+		COMMENT by {@link #markForBlockUpdate}, grouped by their chunk y-section.
```
```suggestion
		COMMENT by {@link #markForBlockUpdate}, grouped by their chunk Y section.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 2 z
-	METHOD method_18686 getX (J)I
+	METHOD method_18686 unpackX (J)I
+		COMMENT Gets the chunk section x-coordinate from the given packed chunk section coordinate.
```
```suggestion
		COMMENT Gets the chunk section X coordinate from the given packed chunk section coordinate.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 sectionCoord
-	METHOD method_18689 getY (J)I
+	METHOD method_18689 unpackY (J)I
+		COMMENT Gets the chunk section y-coordinate from the given packed chunk section coordinate.
```
```suggestion
		COMMENT Gets the chunk section Y coordinate from the given packed chunk section coordinate.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 packed
-	METHOD method_18690 getZ (J)I
+	METHOD method_18690 unpackZ (J)I
+		COMMENT Gets the chunk section z-coordinate from the given packed chunk section coordinate.
```
```suggestion
		COMMENT Gets the chunk section Z coordinate from the given packed chunk section coordinate.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 	METHOD method_18692 toChunkPos ()Lnet/minecraft/class_1923;
-	METHOD method_18693 withZeroZ (J)J
+	METHOD method_18693 withZeroY (J)J
+		COMMENT Gets the packed chunk section coordinate at y=0 for the same chunk as
```
```suggestion
		COMMENT Gets the packed chunk section coordinate at Y=0 for the same chunk as
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 center
 		ARG 1 radius
+	METHOD method_30551 unpackLocalX (S)I
+		COMMENT Gets the local x-coordinate from the given packed local position.
```
```suggestion
		COMMENT Gets the local X coordinate from the given packed local position.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT @see #packLocal
+		ARG 0 packedLocalPos
+	METHOD method_30552 unpackLocalY (S)I
+		COMMENT Gets the local y-coordinate from the given packed local position.
```
```suggestion
		COMMENT Gets the local Y coordinate from the given packed local position.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT @see #packLocal
+		ARG 0 packedLocalPos
+	METHOD method_30553 unpackLocalZ (S)I
+		COMMENT Gets the local z-coordinate from the given packed local position.
```
```suggestion
		COMMENT Gets the local Z coordinate from the given packed local position.
```

shartte replied:
Is this actually correct? I couldn't find any resource that spells x-,y-,z-coordinate without the hyphen.

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT @see #packLocal
+		ARG 0 packedLocalPos
+	METHOD method_30554 unpackBlockX (S)I
+		COMMENT Gets the world x-coordinate of the given local position within this chunk section.
```
```suggestion
		COMMENT Gets the world X coordinate of the given local position within this chunk section.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT @see #packLocal
+		ARG 1 packedLocalPos
+	METHOD method_30555 unpackBlockY (S)I
+		COMMENT Gets the world y-coordinate of the given local position within this chunk section.
```
```suggestion
		COMMENT Gets the world Y coordinate of the given local position within this chunk section.
```

haykam821 commented on mappings/net/minecraft/util/math/ChunkSectionPos.mapping:
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
+		COMMENT @see #packLocal
+		ARG 1 packedLocalPos
+	METHOD method_30556 unpackBlockZ (S)I
+		COMMENT Gets the world z-coordinate of the given local position within this chunk section.
```
```suggestion
		COMMENT Gets the world Z coordinate of the given local position within this chunk section.
```

shartte commented:
I've unified uses of x-coordinate, y-coordinate and z-coordinate and changed it to "retainLighting" over "skipLightingUpdates".
Also changed some uses of "chunk y-section" to just use "vertical chunk section" which is clearer.

modmuss50 commented:
conflicts and then this is good to go. thanks.

shartte committed:
Mapped various aspects of sendings chunks to players and ChunkSectionPos.

shartte commented:
Okay, all fixed
